rlog/2024-12-23-de-mls.mdx

```diff
@@ -0,0 +1,252 @@
+- Scalable
+- End-to-end encrypted (E2EE) 
+- FS and PCS provided
+- Ethereum authenticated 
+
+## Background
+
+### MLS
+
+The Message Layer Security (MLS) protocol offers scalable and secure group messaging protocol by organizing participants into a cryptographic tree structure, enabling efficient operations like adding or removing members with logarithmic time complexity relative to the group size. MLS provides strong security guarantees, including FS and PCS. 
+
+MLS assumes that we have the two services are provided:
+
+   *  An Authentication Service (AS): It enables group members to
+      authenticate the credentials presented by other group members.
+
+   *  A Delivery Service (DS) that routes MLS messages among the
+      participants in the protocol in the correct order and manage the keyPackage of the users where the keyPackage is the object in MLS that each user can securely generate the latest state of keys in the group.  
+      
+Despite its scalability, MLS has a notable limitation: it is inherently designed for server-based federated architectures for delivery service (DS), even when the servers themselves don't need to be trusted. To achieve a decentralized protocol, the functionality of DS must be reimagined to eliminate reliance on a central server while preserving the protocol's security properties. Thus, we proposed decentralized MLS (de-MLS), leveraging Waku nodes as peer-to-peer communication protocols to eliminate reliance on centralized servers.
+
+Lastly, MLS operates on an epoch-based model, where group state changes (e.g., adding/removing users or key refreshes) occur between epochs that are always required to be conducted by a single entity. For example, if a user is removed in epoch `E`, the rest of the group members generate a new key in epoch `E + 1` by passing the new entropy. The removed user cannot decrypt messages sent after epoch `E + 1`.
+
+### Waku
+
+Waku is a peer-to-peer communication network designed to facilitate private, secure, and censorship-resistant communication by eliminating centralized intermediaries. Built on the Gossipsub protocol, Waku operates as a messaging layer where users join groups by subscribing to specific pub-sub topics. These topics define the scope and nature of the messages exchanged, allowing for adaptable communication patterns. The Waku network is composed of multiple interconnected Waku nodes.
+
+## de-MLS 
+
+Decentralized MLS (de-MLS) is a peer-to-peer SGM protocol compatible with any P2P messaging application for the DS, with Waku selected as the messaging protocol. Decentralization is achieved not only at the DS level but also within the authentication service (AS). Multiple special nodes in the group serve as authorized identities to authenticate users before they join or are removed from the group.
+
+de-MLS provides two different user management configurations, both utilizing the Waku protocol for DS:
+
+1. **Single Admin Over a Single Waku Node**: 
```

Comment:
Does the number of waku node matter?

Reply:
hm, you're right, I think it doesn't matter